WikidataItem.py

```diff
@@ -0,0 +1,145 @@
+
+    def make_qualifier_startdate(self, value):
+        prop_item = self.props["start_time"]
+        value_dic = utils.date_to_dict(value, "%Y-%m-%d")
+        value_pwb = pywikibot.WbTime(year=value_dic["year"], month=value_dic[
+                                     "month"], day=value_dic["day"])
+        return self.wdstuff.Qualifier(prop_item, value_pwb)
+
+    def add_statement(self, prop_name, value, quals=None, ref=None):
+        """
+        """
+        base = self.wd_item["statements"]
+        prop = self.props[prop_name]
+        wd_claim = self.make_pywikibot_item(value, prop)
+        statement = self.make_statement(wd_claim)
+        base.append({"prop": prop,
+                     "value": statement,
+                     "quals": quals,
+                     "ref": ref})
+
+    def make_stated_in_ref(self, value, pub_date, ref_url=None):
+        item_prop = self.props["stated_in"]
+        published_prop = self.props["publication_date"]
+        pub_date = utils.date_to_dict(pub_date, "%Y-%m-%d")
+        timestamp = pywikibot.WbTime(year=pub_date["year"], month=pub_date[
+                                     "month"], day=pub_date["day"])
+        source_item = self.wdstuff.QtoItemPage(value)
+        source_claim = self.wdstuff.make_simple_claim(item_prop, source_item)
+        if ref_url:
+            ref_url_prop = self.props["reference_url"]
+            ref_url_claim = self.wdstuff.make_simple_claim(
+                ref_url_prop, ref_url)
+            ref = self.wdstuff.Reference(
+                source_test=[source_claim, ref_url_claim],
+                source_notest=self.wdstuff.make_simple_claim(
+                    published_prop, timestamp))
+        else:
+            ref = self.wdstuff.Reference(
+                source_test=[source_claim],
+                source_notest=self.wdstuff.make_simple_claim(
+                    published_prop, timestamp)
+            )
+        return ref
+
+    def associate_wd_item(self, wd_item):
+        """Associate the data object with a Wikidata item."""
+        if wd_item is not None:
+            self.wd_item["wd-item"] = wd_item
+            print("Associated WD item: ", wd_item)
+
+    def add_label(self, language, text):
+        """
+        Add a label in a specific language.
+
+        :param language: code of language, e.g. "fi"
+        :param text: content of the label
+        """
+        base = self.wd_item["labels"]
+        base.append({"language": language, "value": text})
+
+    def add_description(self, language, text):
+        """
+        Add a description in a specific language.
+
+        :param language: code of language, e.g. "fi"
+        :param text: content of the description
+        """
+        base = self.wd_item["descriptions"]
+        base.append({"language": language, "value": text})
+
+    def construct_wd_item(self):
+        """Create the empty structure of the data object."""
+        self.wd_item = {}
```

Comment:
The more I look at this the more I feel that this should also be an object. That way chunks of what is done in Uploader.py could actually be methods of that class instead. Comparing to [WFD-import](https://github.com/lokal-profil/WFD_import/blob/master/WFD/WFDBase.py#L97) where I end up implementing several similar methods we might even want to create that class over in WikidataStuff.

I would suggest we look into this as a follow-up task though.